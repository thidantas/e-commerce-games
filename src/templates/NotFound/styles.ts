'use client'

import styled from 'styled-components'

import { Container } from 'components/Container'

export const WrapperContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
`
