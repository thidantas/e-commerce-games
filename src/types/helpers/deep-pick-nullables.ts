export type DeepPickNullable<T, K extends readonly unknown[]> = K extends [
  infer First,
  ...infer Rest
]
  ? First extends keyof T
    ? NonNullable<T[First]> extends infer U
      ? Rest extends []
        ? U
        : Rest extends unknown[]
          ? DeepPickNullable<U, Rest>
          : never
      : never
    : never
  : T
