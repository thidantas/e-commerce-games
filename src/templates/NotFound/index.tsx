import Empty from 'components/Empty'

import * as S from './styles'

const NotFound = () => (
  <S.WrapperContainer>
    <Empty
      title="404: Not found"
      description="Ops... this page does not exist. Go back to the store and enjoy our game offers!"
      hasLink
    />
  </S.WrapperContainer>
)

export default NotFound
