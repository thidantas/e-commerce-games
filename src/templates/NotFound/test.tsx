import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import NotFound from '.'

describe('NotFound', () => {
  it('should render a empty component', () => {
    const { container } = renderWithTheme(<NotFound />)

    expect(
      screen.getByRole('heading', { name: /404: Not found/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
