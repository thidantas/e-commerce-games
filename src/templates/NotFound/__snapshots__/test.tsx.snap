// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render a empty component 1`] = `
.c6 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0.8rem;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: 0;
  border-radius: 0.4rem;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c3 {
  max-width: 100%;
}

.c4 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c5 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c0 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
}

<div
  class="c0 c1"
>
  <main
    class="c2"
  >
    <img
      alt="A gamer in a couch playing videogame"
      class="c3"
      role="image"
      src="/img/empty.svg"
    />
    <h2
      class="c4"
    >
      404: Not found
    </h2>
    <p
      class="c5"
    >
      Ops... this page does not exist. Go back to the store and enjoy our game offers!
    </p>
    <a
      href="/"
    >
      <button
        class="c6"
      >
        <span>
          Go back to store
        </span>
      </button>
    </a>
  </main>
</div>
`;
